ddl/partition.go

```diff
@@ -1470,7 +1470,11 @@ func onAlterTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
 	}
 
 	for i, rule := range rules {
-		rule.ID = fmt.Sprintf("%d_%d_%d_%d_%d", job.SchemaID, tblInfo.ID, partitionID, job.ID, i)
+		if rule.Action == placement.RuleOpDel {
+			rule.ID = fmt.Sprintf("%d_t%d_p%d_%s", job.SchemaID, tblInfo.ID, partitionID, rule.Role)
```

Comment:
Do we need to add `i` to `rule.ID` like `RuleOpAdd`?

Reply:
Nope. `jobID` and `i` are used to create unique ID for new rules. As for deletion, you will never know the actual ID of the rules on PD. Because that ID is mixed with random `JobID` and `i`. So deletion here is deleting by prefix.

Reply:
Gotcha.